utils/utils.ts

```diff
@@ -4,30 +4,81 @@
 
 interface IOptions {
   text: string;
-  options: Array<{text:string; correct: boolean}>;
+  options: Array<{ text: string; correct: boolean }>;
 }
 
-export function grade(optionsQuestion: Record<string, boolean>[], answers: IOptions[]) {
-  let notaAcumulada = 0;
-  let allQuestionsRight = true
+export function grade(
+  optionsQuestion: Record<string, boolean>[],
+  answers: IOptions[]
+) {
+  let finalGrade = 0;
+  let allQuestionsRight = true;
 
-  if (optionsQuestion.length !== answers.length) { return "error" }
+  console.log("user", optionsQuestion);
+  console.log("answers", answers);
+
+  if (optionsQuestion.length !== answers.length) {
+    return "error";
+  }
 
   for (let index = 0; index < optionsQuestion.length; index++) {
-    const answerUser = Object.values(optionsQuestion[index])
-    const answerServer = Object.values(answers[index].options)
+    const answerUser = Object.values(optionsQuestion[index]);
+    const answerServer = Object.values(answers[index].options);
 
     for (let indexObject = 0; indexObject < answerUser.length; indexObject++) {
       if (answerUser[indexObject] !== answerServer[indexObject].correct) {
-        allQuestionsRight = false
+        allQuestionsRight = false;
```

Comment:
remember you need to reset this allQuestionsRight for the next iteration if you set this false